File: Official Windows Platform Sample/Windows Phone 8.1 samples/111541-Windows Phone 8.1 samples/Raw notifications sample/C++/SampleBackgroundTask/BackgroundTask.cpp
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO
// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//
// Copyright (c) Microsoft Corporation. All rights reserved

#include "pch.h"
#include "BackgroundTask.h"

using namespace BackgroundTasks;
using namespace Windows::ApplicationModel::Background;
using namespace Windows::Networking::PushNotifications;
using namespace Windows::Storage;

void SampleBackgroundTask::Run(IBackgroundTaskInstance^ taskInstance)
{
    // Get the background task details
    auto settings = ApplicationData::Current->LocalSettings;
    auto taskName = taskInstance->Task->Name;

    // Store the content received from the notification so it can be retrieved from the UI
    auto notificationDetails = dynamic_cast<IRawNotification^>(taskInstance->TriggerDetails);
    settings->Values->Insert(taskName, notificationDetails->Content);
}

File: Official Windows Platform Sample/Windows 8.1 Store app samples/[JavaScript]-Windows 8.1 Store app samples/Raw notifications sample/JavaScript/Shared/js/backgroundTask.js
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO
// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//
// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    // Get the background task details
    var backgroundTask = Windows.UI.WebUI.WebUIBackgroundTaskInstance.current;
    var settings = Windows.Storage.ApplicationData.current.localSettings;
    var taskName = backgroundTask.task.name;

    console.log("Background task \"" + taskName + "\" starting...");

    // Store the content received from the notification so it can be retrieved
    // from the UI.
    var notificationDetails = backgroundTask.triggerDetails;
    settings.values[taskName] = notificationDetails.content;

    console.log("Background \"" + taskName + "\" completed!");

    // Close the instance running the task
    close();
})();
